<a href="https://colab.research.google.com/github/rohansiddam/Python-Journey/blob/main/018%20-%20Project%2018%20(Lists%20%26%20Conversions%20%26%20Loops).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Instructions

#### Goal of the Project

This project is designed for you to practice and solve the activities that are based on the concepts covered in the following lessons:

1. While Loop, Data-Type Conversion and Conditional Statements.

2. Python Lists (`for` Loop).


---

#### Getting Started:

1. Click on this link to open the Colab file for this project.

    https://colab.research.google.com/drive/1n8xEPUsQUXNyXkrYn7WJKa9qFagtUWTW?usp=sharing

2. Create a duplicate copy of the Colab file as described below.

  - Click on the **File menu**. A new drop-down list will appear.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/lesson-0/0_file_menu.png' width=500>

  - Click on the **Save a copy in Drive** option. A duplicate copy will get created. It will open up in the new tab on your web browser.

  <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/lesson-0/1_create_colab_duplicate_copy.png' width=500>

3. After creating the duplicate copy of the notebook, please rename it in the **YYYY-MM-DD_StudentName_Project18** format.

4. Now, write your code in the prescribed code cells.


---

### Activities


  

#### Activity 1: Last Digit Ultimate

Your job is to create a function, that takes 3 numbers: `num1, num2, num3` and returns `True` if:

 (last digit of `num1`) * (the last digit of `num2`) = the last digit of `num3`.

**Examples**
```
1. last_dig(25, 21, 125) ➞ True
# The last digit of 25 is 5, the last digit of 21 is 1,
# last digit of 125 is 5, and the last digit of 5*1 = 5,
# which is equal to the last digit of 125 i.e. 5

2. last_dig(55, 226, 5190) ➞ True
# The last digit of 55 is 5, the last digit of 226 is 6,
# last digit of 5190 is 0, and the last digit of 5*6 = 30 i.e. 0,
# which is equal to the last digit of 5190 i.e. 0

3. last_dig(12, 215, 2142) ➞ False
# The last digit of 12 is 2, the last digit of 215 is 5,
# last digit of 2142 is 2, and the last digit of 2*5 = 10 is 0,
# which is not equal to the last digit of 2142 i.e. 2
```


In [ ]:
# Write your solution here:
def last_dig(num1, num2, num3):

  if int(num1[(len(str(num1)))-1]) * int(num2[(len(str(num2)))-1]) == int(num3[(len(str(num3)))-1]):
    return True
  else:
    return False


---

#### Activity 2: Pentagonal Number

Given a number $num$ as a positive integer, calculate how many dots exist in a pentagonal shape around the centre dot on the $Nth$ iteration.

In the image below you can see in the first iteration there is only a single dot. On the second, there are 6 dots. On the third, there are 16 dots, and on the fourth there are 31 dots.

<img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/pengatonal-number.png' width=500>

**Constraint**

$num > 0$

**Sample Input 1**
```
2
```

**Sample Output 1**
```
6
```

**Sample Input 2**
```
3
```

**Sample Output 2**
```
16
```


In [ ]:
# Write your solution here
num = int(input("Enter in a number:"))

if num == 1:
  print("1")
else:
  print(((5 * num * num) - (5 * num) + 2) // 2)

Enter in a number:9
181


**Q**: How many dots are there in the 9th iteration?

**A**: 181

---

### Submitting the Project:

1. After finishing the project, click on the **Share** button on the top right corner of the notebook. A new dialog box will appear.

  <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/2_share_button.png' width=500>

2. In the dialog box, make sure that '**Anyone on the Internet with this link can view**' option is selected and then click on the **Copy link** button.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/3_copy_link.png' width=500>

3. The link of the duplicate copy (named as **YYYY-MM-DD_StudentName_Project18**) of the notebook will get copied

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/4_copy_link_confirmation.png' width=500>

4. Go to your dashboard and click on the **My Projects** option.
   
   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/5_student_dashboard.png' width=800>

  <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/6_my_projects.png' width=800>

5. Click on the **View Project** button for the project you want to submit.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/7_view_project.png' width=800>

6. Click on the **Submit Project Here** button.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/8_submit_project.png' width=800>

7. Paste the link to the project file named as **YYYY-MM-DD_StudentName_Project18** in the URL box and then click on the **Submit** button.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/9_enter_project_url.png' width=800>

---